src/query.rs

```diff
@@ -931,7 +939,12 @@ mod tests {
         assert!(results.len() >= 1);
 
         for part in results {
-            assert!(part.Caption.contains("Partition #"));
+            assert!(
+                part.Caption
+                    .chars()
+                    .fold(0, |acc, x| if x == '#' { acc + 1 } else { acc })
```

Comment:
Maybe filter+count would be more readable? 
Also, could add a comment with the expected outputs, so that the check makes more sense?

Reply:
Done, check now